crawl-ref/source/dungeon.cc

```diff
@@ -6333,15 +6305,36 @@ static void _stock_shop_item(int j, shop_type shop_type_,
             // Spec shops are allowed to stock trash and may intentionally do so
             if (on_list)
                 break;
-            const bool is_trash = _item_is_trash(env.item[item_index]);
-
-            if (!no_trash && level_number > 10 && is_trash)
-                no_trash = x_chance_in_y(level_number - 10, level_number - 6);
+            item_def& item = env.item[item_index];
+            bool is_trash = _item_is_trash(item);
+            const bool boring_consumable =
+                consumable_rarity(item) >= RARITY_RARE;
+            // Start trashing after Orc:$ depth; starts at a 3/20 probability,
+            // jumps to 50% at Depths:1, 90% by Zot:$.
+            if (!no_trash && level_number > 12 && (is_trash || boring_consumable))
```

Comment:
This comment isn't quite correct. (absdepth-10)/20 gives the following probabilities

30% on Shoals:1/Snake:1/Elf:1 (absdepth 16)
50% on Vaults:1 (absdepth 20)
65% on Depths:1 (absdepth 23)
100% by Zot:3 (absdepth 30)

These probabilities do seem much more reasonable to me now, though.

Reply:
Was adjusting the numbers quite a bit and probably wrote the comment before the last adjustments :) but yeah, these numbers seem quite reasonable.